### ISSUE TITLE
Custom Tab notification never goes away when app is swiped away (or force closed)

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/focus-android#3821]
## Steps to reproduce
1. Open custom tab via third party app (I use [Hews][0] as my test app).
2. Observe the persistent notification for Focus in the notification drawer.
2. Go to app switcher mode, and swipe away the third party app.
3. Observe that persistent notification is _still_ in the notification drawer.

### Expected behavior
- The persisted notification is gone when the app with the custom tag is destroyed.

### Actual behavior
- The persistent notification doesn't go away.
- You have to open the Focus app itself (you can't see any custom tab content there), and then swipe it away to have it disappear.

Video of the issue: https://giant.gfycat.com/TerribleNegligibleImperialeagle.webm

### Device information

* Android device: Pixel 2 (Android 9)
* Focus version: 8.0 (Build #323031233 :lizard: 63.0.20181011200118)

[0]: https://play.google.com/store/apps/details?id=com.leavjenn.hews

**Comment by jonalmeida** [comment]
I should also add for clarity, that the expected behaviour is how the notification used to work prior to this. It gave me context that there was a session still open which had my browsing state set.

**Comment by colintheshots** [comment]
From our discussion last night on this issue... I don't think we have an option as long as our architecture requires the SessionNotificationService running in the foreground. If the notification service is stopped, our code doesn't persist sessions or fields to disk. It would take a lot of refactoring to make the app function without the service while keeping your session. Without the foreground notification and service or a major refactor to persist everything we need, users would lose their sessions regularly.

**Comment by ekager** [comment]
Right, so @jonalmeida , has this behavior changed in recent Nightlies or you're just mentioning a point of confusion?

**Comment by jonalmeida** [comment]
@ekager like I mentioned above (https://github.com/mozilla-mobile/focus-android/issues/3821#issuecomment-435030825) this behaviour has changed. 

Swiping the third party app which had Focus as a custom tab, would dismiss this notification (ending the service). Now, it stays there forever until you actively open Focus from the launcher and then swipe the Focus app away.

I believe that this _might_ be the root cause of #3811 as well.

**Comment by ekager** [comment]
Did a bit of investigation: 

onTaskRemoved in SessionNotificationService is not called when a custom tab's app is swiped away. This may have been introduced when we changed our launchMode?

We also start/stop our notification in NotificationSessionObserver, but custom tabs aren't observed here. This also could have been introduced when we switched to the AC Session Manager ( #3179 ). 

We could add a check in onDestroy in CustomTabActivity if the sessionManager has no sessions, stop the SessionNotificationService, but I also found a related bug. If you open a custom tab, and click the notification to erase, you're brought to the Focus homescreen which we never want to show in a custom tab.

I think we need to rethink this SessionNotificationService and NotificationSessionObserver as we want it to relate to custom tabs.
